worlds/pokemon_rb/pokemon.py

```diff
@@ -1,9 +1,17 @@
 from copy import deepcopy
+import typing
+
+from worlds.Files import APTokenTypes
+
 from . import poke_data, logic
 from .rom_addresses import rom_addresses
 
+if typing.TYPE_CHECKING:
+    from . import PokemonRedBlueWorld
+    from .rom import PokemonRedProcedurePatch, PokemonBlueProcedurePatch
+
 
-def set_mon_palettes(world, random, data):
+def set_mon_palettes(world: "PokemonRedBlueWorld", patch: typing.Union["PokemonRedProcedurePatch", "PokemonBlueProcedurePatch"]):
```

Comment:
my understanding is quoted typings need to be fully quoted, also | unions are preferred over typing.Union in py3.10+
```suggestion
def set_mon_palettes(world: "PokemonRedBlueWorld", patch: "PokemonRedProcedurePatch | PokemonBlueProcedurePatch"):
```

Reply:
I didn't realize it was intended to quote the whole thing rather than just the types that need it. But I think the pipe operator is finicky with quoted types and `typing.Union` is still preferred.

Reply:
> pipe operator is finicky with quoted types

The "quote the whole thing" is the defined behavior, so I would argue that's not that finicky... it works as documented

> Note The | operand cannot be used at runtime to define unions where one or more members is a forward reference. For example, int | "Foo", where "Foo" is a reference to a class not yet defined, will fail at runtime. For unions which include forward references, present the whole expression as a string, e.g. "int | Foo".

https://docs.python.org/3/library/stdtypes.html#types-union

and

> typing.Union is still preferred

In general, that's not the case now that we're on 3.10+. "To define a union, use e.g. Union[int, str] or the shorthand int | str. Using that shorthand is recommended." https://docs.python.org/3/library/typing.html#typing.Union
Maybe forward references is an exception to that rule though?

Reply:
I suppose I (or someone I listened to) was under the impression that forward references and pipes weren't fully functional/compatible, warranting the use of `typing.Union`, but I guess it only means that the quotes have to surround the whole thing.